### DISCUSSION TITLE
Trouble understanding where to start

### DISCUSSION THREAD

**Discussion by Eduard-Hasa** [pointfreeco/swift-composable-architecture#2854] [Q&A] [answered]
I recently discovered this repo and instantly became a paid subscriber. Absolutely love the material but having trouble with understanding what advice/content is still relevant. It seems everything in this space has been drastically changing (which is amazing) but it's confusing me on what content I can learn from and what I should completely ignore. I am referring to paid videos and GitHub repo content/examples.

I am a senior react native/node/typescript dev that is picking up swift & SwiftUI as quickly as possible. I have tons of experience with redux toolkit and their style guide. 

My end goal is to get an understanding how to do things like an e-commerce app with a login screen, products list screen, and orders screen which have simple api calls to my node js rest api. In redux toolkit it handled the cacheing for us so also a bit confused how we would achieve that functionality in tca

**Comment by mbrandonw** [comment]
Hi @Eduard-Hasa, the best place to start for videos is the [tour of 1.0](https://www.pointfree.co/collections/composable-architecture/composable-architecture-1-0). It shows off the most important aspects of the library, such as navigation, domain modeling, dependencies, side effects, etc.

Then in this very repo there are lots of [case studies and demo apps](https://github.com/pointfreeco/swift-composable-architecture/tree/main/Examples) that show how to solve common problems with the library. Further, the [docs](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/) are quite extensive and describe a lot of the most important ideas in the library.

And then once you are comfortable with the basics of the library and you are curious how/why we built things the way they are, you can check out any of the [past videos](https://www.pointfree.co/collections/composable-architecture) on Point-Free. Those are all deep dives into the kinds of problems we are trying to solve with TCA, with a full explanation of why we settled on the solutions we did.

**Reply by Eduard-Hasa** [reply]
@mbrandonw 

Okay awesome im excited. So the tour of 1.0 has been updated to recommend the latest features and anything I see there I can follow fully or do I need to watch the other sections after to learn the new updates and changes?

I just don't want to watch one section of videos only to have to relearn it in the following sections.

**Reply by mbrandonw** [reply, marked as answer]
Hi @Eduard-Hasa, sorry no the 1.0 tour has not been updated for observation, but it does mention in various parts how things will be simplified by the (then upcoming) observation tools. The tour still has all of the most important concepts, and then to get up-to-date with the newest version of the library you can follow the [migration guides](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/migrationguides).

Also, the app that was built during the 1.0 tour was live refactored to make use of observation during our last [livestream](https://www.pointfree.co/episodes/ep267-point-free-live-observation-in-practice). So, you could watch the 1.0 tour, and then that segment of the livestream to see how to use the observation tools.

We probably will not redo the tour until 2.0, but we may also do something sooner.

**Reply by Eduard-Hasa** [reply]
Keep up the amazing work. Absolutely impressed and will be joining for the next livestream. I am a lifetime subscriber!
